master/internal/stream/trials.go

```diff
@@ -216,3 +280,24 @@ func (ts *TrialFilterMaker) MakeFilter() func(*TrialMsg) bool {
 		return false
 	}
 }
+
+// TrialMakePermissionFilter returns a function that checks if a TrialMsg
+// is in scope of the user permissions.
+func TrialMakePermissionFilter(ctx context.Context, user model.User) (func(*TrialMsg) bool, error) {
+	accessScopeSet, err := AuthZProvider.Get().GetTrialStreamableScopes(ctx, user)
+	if err != nil {
+		return nil, err
+	}
+
+	var permissionFilter func(*TrialMsg) bool
```

Comment:
maybe return from inside the cases, rather than have an explicit variable?

Reply:
(`return func(msg *TrialMsg) bool { return true }`)

Reply:
Sounds good, time to unlearn some of the dogma's of structured programming 🤣